### PR TITLE
Refactor grid editor data access

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request type

 

- Refactoring (no functional changes, no API changes)

## What is the current behavior?

Data access from any given code point is hard, even given a `NatTable` object, as this is actively discouraged by the framework.

## What is the new behavior?

- Calls to `NatTable.getDataValueByPosition(columnPosition, rowPosition)` return the Salt object instead of just the value string.
- Merge display converters into one
- Removes unneeded custom SpanningProvider

## Does this introduce a breaking change?

No.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added **or** the PR is neither a bug fix nor a feature
- [x] Docs have been reviewed and added / updated if needed **or** the PR is neither a bug fix nor a feature
- [x] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR if needed
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [x] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)
- [x] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary **or** there are no new dependencies

---

# Checklist for maintainers

## Releases

- Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- Check that license and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: tests/org.corpus_tools.hexatomic.grid.tests/src/main/java/org/corpus_tools/hexatomic/grid/internal/style/TestSelectionStyleConfiguration.java [hunk 1]
```diff
@@ -1,7 +1,8 @@
-package org.corpus_tools.hexatomic.grid.style;
+package org.corpus_tools.hexatomic.grid.internal.style;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+import org.corpus_tools.hexatomic.grid.internal.style.SelectionStyleConfiguration;
```
Comment (thomaskrause) [inline tests/org.corpus_tools.hexatomic.grid.tests/src/main/java/org/corpus_tools/hexatomic/grid/internal/style/TestSelectionStyleConfiguration.java:5]: This import is not really used.
```suggestion
```

File: tests/org.corpus_tools.hexatomic.grid.tests/src/main/java/org/corpus_tools/hexatomic/grid/internal/bindings/TestFreezeGridBindings.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    fixture = new FreezeGridBindings();
+  }
+
+  /**
+   * Test method for
+   * {@link org.corpus_tools.hexatomic.grid.internal.bindings.FreezeGridBindings#configureUiBindings(org.eclipse.nebula.widgets.nattable.ui.binding.UiBindingRegistry)}.
+   */
+  @Test
+  void testConfigureUiBindings() {
+    // UiBindingRegistry registry = mock(UiBindingRegistry.class);
```
Comment (thomaskrause) [inline tests/org.corpus_tools.hexatomic.grid.tests/src/main/java/org/corpus_tools/hexatomic/grid/internal/bindings/TestFreezeGridBindings.java:40]: Commented-out code line is not really informative
```suggestion
```

**Review by thomaskrause — CHANGES_REQUESTED** [review]
Seems good to me in general but I would really like to remove the "internal" package from the exported packages of the bundle (seem comment).

Also, I have trouble to add new events for only a selected range of token. E.g. for the GUM corpus, In the current master branch, empty cells are rendered for each token and I can add an event for any number of tokens.
Peek 2020-09-14 14-44.
But in this PR, I can only select the whole empty range between other annotations.
Peek 2020-09-14 14-46

File: bundles/org.corpus_tools.hexatomic.grid/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -14,7 +14,8 @@ Require-Bundle: javax.inject,
  org.eclipse.e4.ui.workbench,
  org.eclipse.e4.ui.services,
  org.eclipse.e4.core.di.annotations,
- javax.annotation;bundle-version="1.2.0"
+ javax.annotation;bundle-version="1.2.0",
+ org.corpus-tools.salt-extensions
 Automatic-Module-Name: org.corpus_tools.hexatomic.grid
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
-Export-Package: org.corpus_tools.hexatomic.grid.style
+Export-Package: org.corpus_tools.hexatomic.grid.internal.style;x-internal:=true
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/META-INF/MANIFEST.MF:21]: I agree.

[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/META-INF/MANIFEST.MF]: Fixed in aac461420fae6fc9405b765fb25b78206212d284.

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridEditor.java [hunk 4]
```diff
@@ -228,8 +234,63 @@ private void addTextSelectionDropdown(Composite parent) {
...
+      @Override
+      public void selectionChanged(SelectionChangedEvent event) {
+        IStructuredSelection selection = (IStructuredSelection) event.getSelection();
+        if (selection.size() > 0 && selection.getFirstElement() instanceof STextualDS) {
+          // Dispose select message, once a selection is made, selection can never be null or empty
+          // again.
+          messageLabel.dispose();
+          parent.layout();
+          // Set decoration depending on whether there are tokens in the data source.
+          if (((STextualDS) selection.getFirstElement()).getInRelations().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridEditor.java:266]: Makes sense.

[the remark this answers is not in the document]
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridEditor.java]: Fixed in 3c80fbd3.

**Comment by sdruskat** [comment]
Merged cell issue reported is fixed in https://github.com/hexatomic/hexatomic/pull/218/commits/00350b2aad9a38a7d04beea80d690168ce4897b3.

**Review by thomaskrause — APPROVED** [review]

**Pull request merged**